src/app/routes/api/v3/public/public-forms.routes.ts

```diff
@@ -0,0 +1,103 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { Router } from 'express'
+
+import { rateLimitConfig } from '../../../../config/config'
+import * as PublicFormController from '../../../../modules/form/public-form/public-form.controller'
+import * as EmailSubmissionController from '../../../../modules/submission/email-submission/email-submission.controller'
+import * as EmailSubmissionMiddleware from '../../../../modules/submission/email-submission/email-submission.middleware'
+import * as EncryptSubmissionController from '../../../../modules/submission/encrypt-submission/encrypt-submission.controller'
+import * as EncryptSubmissionMiddleware from '../../../../modules/submission/encrypt-submission/encrypt-submission.middleware'
+import { CaptchaFactory } from '../../../../services/captcha/captcha.factory'
+import { limitRate } from '../../../../utils/limit-rate'
+
+export const PublicFormsRouter = Router()
+
+/**
+ * Send feedback for a public form
+ * @route POST /:formId/feedback
+ * @group forms - endpoints to serve forms
+ * @param {string} formId.path.required - the form id
+ * @param {Feedback.model} feedback.body.required - the user's feedback
+ * @consumes application/json
+ * @produces application/json
+ * @returns 200 if feedback was successfully saved
+ * @returns 400 if form feedback was malformed and hence cannot be saved
+ * @returns 404 if form with formId does not exist or is private
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs
+ */
+PublicFormsRouter.route('/:formId([a-fA-F0-9]{24})/feedback').post(
+  celebrate({
+    [Segments.BODY]: Joi.object()
+      .keys({
+        rating: Joi.number().min(1).max(5).cast('string').required(),
+        comment: Joi.string().allow('').required(),
+      })
+      // Allow other keys for backwards compability as frontend might put
+      // extra keys in the body.
+      .unknown(true),
+  }),
```

Comment:
what do you think about moving this validation middleware into PublicFormController, and exporting both the validation middleware and `handleSubmitFeedback` as an array of middleware instead of separately?

we actually decided to practice this a while ago, but we haven't been very good about enforcing it. it makes sense because `handleSubmitFeedback` assumes that the request has a certain shape, and can only do so safely when the validation middleware comes before it. so the validation and the handler must always go together, so it makes sense to export them as a single unit.

Reply:
do you mean something along the lines of 

```ts
const handleSubmitFeedback = ... (same as current) 
export const handleSubmitFeedbackWithValidation = [celebrate(...), handleSubmitFeedback]
```

Reply:
@mantariksh following up on this - i will collapse 2 other separate endpoints as well 

1. handleEncryptedSubmission
2. handleEmailSubmission 

the only visible middleware should be the rate limiter. This is because we want to always validate captcha and submission parameters before deferring to the respective controllers but limit rate is tweakable and should not be viewed as being crucial to handling the respective submissions.

feel free to correct me if i'm wrong on this!

Reply:
what about `receiveEmailSubmission`?

Reply:
hidden from routes and is exported as part of `handleEmailSubmission`

Reply:
excellent!